inc/admin/class-admin.php

```diff
@@ -168,28 +169,28 @@ public function check_for_event_submissions(){
 							$cat_name = $new_line;
 							$cat_slug = $new_line;
 						}
-						
+
 						if ($i == 0) {
-							if ( $this->termAlreadyExists($rootCategories, 'name', $cat_name, 'id') ) {
-								$parent_id = $this->termAlreadyExists($rootCategories, 'name', $cat_name, 'id');
+							//use isset rather than separate method, simpler.
+							if ( isset($rootCategories[$cat_name]) ) {
+								$parent_id = $rootCategories[$cat_name];
 								$countErrors++;
 							} else {
 								$result = wp_insert_term( $cat_name, $taxonomyActive, array('slug' => $cat_slug) );
 								if ( ! is_wp_error( $result ) ) {
 									$parent_id = isset( $result['term_id'] ) ? $result['term_id'] : '';
-									$rootCategories[] = array('id' => $parent_id, 'name' => $cat_name);
+									$rootCategories[$cat_name] = $parent_id;
```

Comment:
tracking the newly created root terms need to follow same format as wp_list_pluck object.